File: src/Web.Admin/Data/MockUsersData.cs
using BarbeariaSaaS.Web.Admin.Models;
using System.Security.Cryptography;
using System.Text;

namespace BarbeariaSaaS.Web.Admin.Data;

public static class MockUsersData
{
    private static readonly List<User> _users = new()
    {
        // Admin SaaS
        new User
        {
            Id = "admin-001",
            Email = "guelfi@msn.com",
            Name = "Administrador SaaS",
            Role = UserRole.Admin,
            TenantId = "saas-admin",
            IsActive = true,
            CreatedAt = new DateTime(2024, 1, 1)
        }
    };

    private static readonly Dictionary<string, string> _passwords = new()
    {
        { "guelfi@msn.com", "@5ST73EA4x" }
    };

    public static User? ValidateCredentials(string email, string password)
    {
        var user = _users.FirstOrDefault(u => 
            u.Email.Equals(email, StringComparison.OrdinalIgnoreCase));
        
        if (user == null || !user.IsActive)
        {
            return null;
        }
        
        if (!_passwords.TryGetValue(user.Email, out var expectedPassword) || 
            expectedPassword != password)
        {
            return null;
        }
        
        return user;
    }
    
    public static User? GetUserByEmail(string email)
    {
        return _users.FirstOrDefault(u => 
            u.Email.Equals(email, StringComparison.OrdinalIgnoreCase));
    }
    
    public static User? GetUserById(string id)
    {
        return _users.FirstOrDefault(u => u.Id == id);
    }
    
    public static IEnumerable<User> GetUsersByTenant(string tenantId)
    {
        return _users.Where(u => u.TenantId == tenantId);
    }
    
    public static string HashPassword(string password)
    {
        // Simple mock hash - in real app would use BCrypt or similar
        using var sha256 = SHA256.Create();
        var hashedBytes = sha256.ComputeHash(Encoding.UTF8.GetBytes(password + "salt123"));
        return Convert.ToBase64String(hashedBytes);
    }
    
    public static bool VerifyPassword(string password, string hash)
    {
        return HashPassword(password) == hash;
    }
    
    public static void AddUser(User user, string password)
    {
        _users.Add(user);
        _passwords[user.Email] = password;
    }
}

File: src/Web.Desktop/src/app/core/data/mock-users.data.ts
import { User, UserRole } from '../models/auth.models';

export const MOCK_USERS: User[] = [
  // Admin SaaS
  {
    id: 'admin-001',
    email: 'guelfi@msn.com',
    name: 'Administrador SaaS',
    role: UserRole.ADMIN,
    tenantId: 'saas-admin',
    isActive: true,
    createdAt: new Date('2024-01-01')
  },
  
  // Barbearia Demo - Barbeiros
  {
    id: 'barbeiro-001',
    email: 'barbeiro@barbearia.com',
    name: 'João Silva',
    role: UserRole.BARBEIRO,
    tenantId: 'tenant-001',
    isActive: true,
    createdAt: new Date('2024-01-15')
  },
  {
    id: 'barbeiro-002',
    email: 'carlos@barbearia.com',
    name: 'Carlos Santos',
    role: UserRole.BARBEIRO,
    tenantId: 'tenant-001',
    isActive: true,
    createdAt: new Date('2024-01-20')
  },
  
  // Barbearia Demo - Recepcionista
  {
    id: 'recep-001',
    email: 'recepcao@barbearia.com',
    name: 'Maria Oliveira',
    role: UserRole.RECEPCIONISTA,
    tenantId: 'tenant-001',
    isActive: true,
    createdAt: new Date('2024-01-10')
  },
  
  // Clientes Demo
  {
    id: 'cliente-001',
    email: 'cliente@email.com',
    name: 'Pedro Costa',
    role: UserRole.CLIENTE,
    tenantId: 'tenant-001',
    isActive: true,
    createdAt: new Date('2024-02-01')
  },
  {
    id: 'cliente-002',
    email: 'jose@email.com',
    name: 'José Ferreira',
    role: UserRole.CLIENTE,
    tenantId: 'tenant-001',
    isActive: true,
    createdAt: new Date('2024-02-05')
  }
];

export const MOCK_PASSWORDS: { [email: string]: string } = {
  'guelfi@msn.com': '@5ST73EA4x',
  'barbeiro@barbearia.com': 'Barbeiro123!',
  'carlos@barbearia.com': 'Carlos123!',
  'recepcao@barbearia.com': 'Recepcao123!',
  'cliente@email.com': 'Cliente123!',
  'jose@email.com': 'Jose123!'
};

export class MockAuthDatabase {
  static validateCredentials(email: string, password: string): User | null {
    const user = MOCK_USERS.find(u => u.email.toLowerCase() === email.toLowerCase());
    
    if (!user) {
      return null;
    }
    
    if (!user.isActive) {
      return null;
    }
    
    const expectedPassword = MOCK_PASSWORDS[user.email];
    if (expectedPassword !== password) {
      return null;
    }
    
    return user;
  }
  
  static getUserByEmail(email: string): User | null {
    return MOCK_USERS.find(u => u.email.toLowerCase() === email.toLowerCase()) || null;
  }
  
  static getUserById(id: string): User | null {
    return MOCK_USERS.find(u => u.id === id) || null;
  }
  
  static getUsersByTenant(tenantId: string): User[] {
    return MOCK_USERS.filter(u => u.tenantId === tenantId);
  }
  
  static hashPassword(password: string): string {
    // Simple mock hash - in real app would use bcrypt or similar
    return btoa(password + 'salt123');
  }
  
  static verifyPassword(password: string, hash: string): boolean {
    return this.hashPassword(password) === hash;
  }
}